### PR TITLE
Add retry for creating a DB connection

### PR DESCRIPTION
**Pull request by tsachiherman**
## Summary

Ledger service was generating the following error:
```
{"file":"ledgerService.go","function":"github.com/algorand/go-algorand/rpcs.(*LedgerService).ServeHTTP","level":"warning","line":174,"msg":"ServeHTTP : failed to retrieve block 3735696 database is locked","time":"2020-03-15T16:21:05.316959Z"}
```

The root cause was the sqlite driver returning a `database is locked` when a new connection attempt was made. Existing code was already testing extensively for temporary locked database conditions on transaction creation/execution, but not on connection creation.

resolves issue https://github.com/algorand/go-algorand/issues/915

### CODE REVIEW

File: util/db/dbutil.go [hunk 1]
```diff
@@ -209,21 +212,42 @@ func (db *Accessor) Atomic(fn idemFn, extras ...interface{}) (err error) {
 	}
 
 	var tx *sql.Tx
-	ctx := context.Background()
 	var conn *sql.Conn
-	conn, err = db.Handle.Conn(ctx)
+	ctx := context.Background()
+
+	for i := 0; ; i++ {
```
Comment (pzbitskiy) [inline util/db/dbutil.go]: use LoggedRetry?

File: util/db/dbutil.go [hunk 2]
```diff
@@ -145,12 +145,15 @@ func (db *Accessor) Close() {
 // Sends warnings and errors to log.
 func LoggedRetry(fn func() error, log logging.Logger) (err error) {
 	for i := 0; ; i++ {
-		if i > 0 && i%warnTxRetries == 0 {
```
Comment (pzbitskiy) [inline util/db/dbutil.go]: nit: reduce amount of branches with something like this:
```
conn, err = db.Handle.Conn(ctx)
for i := 1; dbretry(err); i++ {
  // logging
  conn, err = db.Handle.Conn(ctx)
}
```

**Comment by justicz** [comment]
Just confirming, were you able to reproduce this locally?

File: util/db/dbutil.go (hunk 2 above)
Comment (tsachiherman) [reply util/db/dbutil.go]: good idea.

File: util/db/dbutil.go (hunk 1 above)
Comment (tsachiherman) [reply util/db/dbutil.go]: I want to have a different info/warn/err message to indicate that it's a connection error. If this becomes persistent, we need to have that information.

**Comment by tsachiherman** [comment]
> Just confirming, were you able to reproduce this locally?

regretfully, no. And it's not due to lack of trying. I suspect it's an external issue, but the information that we currently have isn't enough to back this up. The added info/warn/error on the connection attempt should help us confirm that its a permanent issue caused by a misconfigured installation. ( i.e. maybe somehow the algod lost the ability to make changes to the database ? )

**Pull request merged**